### DISCUSSION TITLE
Question about reading outputs from CLI Tool vs Chrome Extension

### DISCUSSION THREAD

**Discussion by zinkohlaing** [GoogleChromeLabs/ps-analysis-tool#609] [Support Forum] [answered]
Hi,

Technically this might not be a bug, but I am very keen to understand the differences between the output from CLI versus Chrome Extension.

Also, I would appreciate if you can provide the best practices for conducting the cookie audit using Chrome extension.

I am going to this URL for example: https://adviser.tal.com.au/

I am using a Chrome Canary, with everything (history, cookies) cleared, and opened the site for the first time. I just opened the page without doing any interaction.

PSAT extension gives me this result:

![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/13776707/f27c26c5-502c-4928-9016-86c9ccb935f4)

And I use CLI tool to run audit on the same page and got this result:
![image](https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/13776707/3c9683e1-81f0-4652-a78e-3a6e7f9d2317)

I understand that there might be some difference, but shouldn't that be similar at least? How can I explain the differences if I am to do an audit of entire site using CLI tool?

Thank you!

**Comment by milindmore22** [comment, marked as answer]
Hello @zinkohlaing 

Thanks for reaching out to us, We noticed in your screenshots that Chrome DevTools Protocol (CDP) might not be enabled for the PSAT extension.

PSAT relies on CDP for in-depth analysis of your site, You can enable CDP from the [extension settings](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/PSAT-Settings-and-Permissions#enabling-chrome-devtools-protocol-in-psat) or directly from the [PSAT popup](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/PSAT-Extension-Popup), as explained on our [wiki page](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/PSAT-Settings-and-Permissions#enabling-chrome-devtools-protocol-in-psat). An inactive CDP could be why you’re seeing a significant difference between the extension’s and CLI’s cookie analysis.

It’s important to note that there can still be some discrepancies between the CLI and extension, and the reasons for this are detailed on our [wiki page](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/PSAT-Command-Line-Interface#discrepancy-between-cli-and-extension) as well.

**Here are some best practices for conducting a cookie audit using the PSAT Chrome extension:**

**Compare Site Behavior with and Without Third-Party Cookies:**
- Enable and disable third-party cookies in your browser settings.
- Use the PSAT extension to analyze cookies on your site in both scenarios.
- Look for any changes in functionality, especially features that rely on third-party cookies (e.g., social media logins, ad targeting, analytics).

**Simulate User Flows and Check Functionality:**

- Identify key user flows on your site, such as login, checkout, and content sharing.
- Use the PSAT extension to analyze cookies involved in these user flows.
- Manually test these functionalities to see if they are impacted by cookie blocking.

**Explore Alternatives for Impacted Features:**

- If user flows are broken due to blocked third-party cookies, research alternative solutions, [Explore alternative](https://developers.google.com/privacy-sandbox/3pcd) to migrate from third-party cookies

For future references if you have any further questions related to PSAT please contact us on [support forums](https://github.com/GoogleChromeLabs/ps-analysis-tool/discussions/new?category=support-forum)
